framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -438,17 +478,18 @@ def localFinalizeActualSampling(self,jobObject,model,myInput):
     # for some reason, Ensemble Model doesn't preserve this information, so wrap this debug in a try:
     try:
       prefix = jobObject.getMetadata()['prefix']
-      self.raiseADebug('Collected sample "{}"'.format(prefix))
     except TypeError:
       prefix = ''
     self.raiseADebug('Collected sample "{}"'.format(prefix))
 
     # TODO REWORK move this whole piece to Optimizer base class as much as possible
+    # TODO what if there's no solution export?  Our "status" and etc fail badly!
```

Comment:
It fails badly.  I'm going to make SolutionExport required for now.

To be honest, you get nothing useful out of the optimizer if you don't use a solution export, so I can't think of a case where you shouldn't have it anyway.